src/window.cpp

```diff
@@ -2381,10 +2381,11 @@ static EventState HandleViewportScroll()
 	Point delta;
 	if (scrollwheel_scrolling) {
 		/* We are using scrollwheels for scrolling */
-		delta.x = _cursor.h_wheel;
-		delta.y = _cursor.v_wheel;
-		_cursor.v_wheel = 0;
-		_cursor.h_wheel = 0;
+		delta.x = (int)_cursor.h_wheel;
+		delta.y = (int)_cursor.v_wheel;
+		float temp;
+		_cursor.v_wheel = std::modf(_cursor.v_wheel, &temp);
+		_cursor.h_wheel = std::modf(_cursor.h_wheel, &temp);
```

Comment:
I don't really understand what this is doing. I think it's removing the integer part of the value, but I don't know why.

Reply:
Please add this comment to the code.

Can I also ask you to use C++-style casts `static_cast<type>(x)` instead of C-style casts `(type)x`, both here and the other places. (We're slowly switching style of existing casts, but new stuff should be C++ casts)

Reply:
Sure, should be resolved now.